tasker/executors/parallel_executor.py

```
@@ -74,8 +76,22 @@ def _execute_single_task_with_retry_core(task, master_timeout, retry_config, con
                 # Real failure - retry makes sense
                 next_attempt_display = f".{attempt + 2}" if retry_config else ""
                 executor_instance.log(f"Task {parent_task_id}-{task_id}{retry_display}: FAILED - will retry as Task {parent_task_id}-{task_id}{next_attempt_display} in {retry_delay}s")
-                if not executor_instance.dry_run:
-                    time.sleep(retry_delay)
+                if not executor_instance.dry_run and retry_delay > 0:
+                    # Use non-blocking sleep for retry delay
+                    retry_completed_event = threading.Event()
+
+                    def retry_callback():
+                        retry_completed_event.set()
```

Ruff (0.13.2)

Function definition does not bind loop variable `retry_completed_event`

(B023)